src/script.js

```diff
@@ -0,0 +1,273 @@
+      ScriptHelper.REDIRECT_URI,
+    );
+    const code = await this.#getCode(authUrl);
+    await authenticator.handleCallback({ code });
+    this.#logger.info(`Refresh Token: ${authenticator.refreshToken}`);
+  }
+
+  async confirm(message) {
+    const response = await this.#ask(`${message} [y/Y]:`);
+    if (response.toLowerCase() !== 'y') {
+      process.exit(1);
+    }
+  }
+
+  async confirmOptions() {
+    this.#reader = createInterface({
+      input: process.stdin,
+      output: new LogWriter(this.#logger),
+    });
+    this.#logger.info(`${this.#config.name} - Confirm Options`);
+    await eachSeries(this.#config.options, async (op) => this.#confirmOption(op));
+    await this.confirm('Ready to proceed?');
+  }
+
+  /**
+   * @param {string} question
+   * @returns {Promise<string>}
+   */
+  #ask(question) {
+    return new Promise((resolve) => {
+      this.#reader.question(question, (response) => {
+        resolve(response);
+        this.#reader.pause();
+      });
+    });
+  }
+
+  /**
+   * @returns {boolean} true, if there is a state file
+   */
+  batchHasSavedState() {
+    const stateFile = this.#settings[PN_STATE_FILE] || '.state.json';
+    return existsSync(stateFile);
+  }
+
+  /**
+   * @param {Option} option
+   * @returns {Promise<void>}
+   */
+  async #confirmOption(option) {
+    const defaultVal = process.env[option.env];
+    const response = await this.#ask(
+      `${option.display || option.name} [${defaultVal}]: `,
+    );
+    if (response && response.trim() !== '') {
+      this.#settings[option.name] = response;
+    }
+    this.#settings[option.name] = defaultVal;
+  }
+
+  /**
+   * Get the code from an Oauth response
+   * @param {string} authUrl the url for the user to login
+   * @returns {Promise<string>} the code
+   */
+  async #getCode(authUrl) {
+    const sockets = new Set();
+    const server = createServer();
+    server.listen(3000, () => {
+      this.#logger.info(`Please open: ${authUrl}`);
+    });
+    server.on('connection', (socket) => {
+      sockets.add(socket);
+      server.once('close', () => {
+        sockets.delete(socket);
+      });
+    });
+    return new Promise((resolve) => {
+      server.on('request', (req, res) => {
+        const url = new URL(`${ScriptHelper.REDIRECT_URI}${req.url}`);
+        const code = url.searchParams.get('code');
+        if (code) {
+          res.statusCode = 200;
+          res.write('Code recieved successfully');
+          res.end();
+          resolve(code);
+        }
+      });
+    }).finally(() => {
+      server.close();
+      sockets.forEach((socket) => socket.destroy());
+    });
+  }
+
+  getIngestorClient() {
+    const jobId = randomUUID();
+    return new IngestorClient({
+      url: this.#settings[PN_INGESTOR_URL],
+      apiKey: this.#settings[PN_INGESTOR_API_KEY],
+      companyId: this.#settings[PN_COMPANY_ID],
+      jobId,
+      log: this.#logger,
+      spaceId: this.#settings[PN_SPACE_ID],
```

Comment:
I couldn't see how `#settings` is set, but my only thought would be to eventually just pass through whichever values are passed to the script, rather than explicitly setting a few. That might make things more flexible if, for example, an ingestor supports more options than just these. That doesn't necessarily need to be part of this PR, just a potential improvement for the future.